src/Nuget.PackageIndex/SettingsJson.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                jObject = ReadJsonFromFile(_settingsJsonPath);
+            } 
+            else
+            {
+                jObject = GenerateDefaultSettingsJson();
+                WriteJsonToFile(jObject);
+            }
+
+            return jObject;
+        }
+
+        private JObject GenerateDefaultSettingsJson()
+        {
+            var jObject = JObject.Parse(DefaultJsonContent);
+            JArray includePackagePaterns = (JArray)jObject["includePackagePatterns"];
+            foreach(var pattern in _defaultIncludeRules)
+            {
+                includePackagePaterns.Add(pattern);
+            }
+
+            return jObject;
+        }
+
+        private JObject ReadJsonFromFile(string filePath)
+        {
+            JObject result = null;
+
+            try
+            {
+                var fileContent = _fileSystem.FileReadAllText(filePath);
+                result = JObject.Parse(fileContent);
+            }
+            catch (Exception e)
+            {
+                Debug.Write(e.ToString());
+            }
+
+            return result;
+        }
+
+        private void WriteJsonToFile(JObject json)
+        {
+            try
+            {
+                if (_fileSystem.FileExists(_settingsJsonPath))
+                {
+                    _fileSystem.FileDelete(_settingsJsonPath);
+                }
+
+                using (FileStream fs = File.Open(_settingsJsonPath, FileMode.OpenOrCreate))
+                using (StreamWriter sw = new StreamWriter(fs))
+                using (JsonWriter jw = new JsonTextWriter(sw))
+                {
+                    jw.Formatting = Formatting.Indented;
+
+                    json.WriteTo(jw);
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.Write(e.ToString());
+            }
+        }
+
+        private IEnumerable<string> ReadJsonArray(string nodeName)
+        {
+            try
+            {
+                var settingsJson = InitializeSettingsJson();
+                if (settingsJson != null)
+                {
+                    var array = settingsJson[nodeName] as JArray;
+                    if (array != null)
+                    {
+                        return array.Select(x => x.ToString());
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.Write(e.ToString());
+            }
+
+            return null;
+        }
+
+        private List<string> GetIncludePackagePatterns()
+        {
+            List<string> result = null;
+
+            var includeRulesFromJsonFile = ReadJsonArray("includePackagePatterns");
+            if (includeRulesFromJsonFile != null)
+            {
+                result = new List<string>(includeRulesFromJsonFile);
+            }
+
+            return result ?? new List<string>(_defaultIncludeRules);
+        }
+
+        #region ISettingsJson 
```

Comment:
well we do use regions in valid cases like here .. its just simpler to find where interface methods/properties are defined in the class among other methods